src/ssmDocument/ssm/ssmServer.ts

```diff
@@ -0,0 +1,537 @@
+ */
+
+/*!
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+// tslint:disable: no-inferred-empty-object-type
+
+import { JsonLS, getLanguageServiceSSM as getLanguageService } from 'aws-ssm-document-language-service'
+
+import {
+    createConnection,
+    Disposable,
+    DocumentRangeFormattingRequest,
+    IConnection,
+    InitializeParams,
+    InitializeResult,
+    NotificationType,
+    RequestType,
+    ServerCapabilities,
+    TextDocuments,
+    TextDocumentSyncKind,
+} from 'vscode-languageserver'
+
+import { posix } from 'path'
+import { clearTimeout, setTimeout } from 'timers'
+import * as URL from 'url'
+import { getLanguageModelCache } from '../../stepFunctions/asl/languageModelCache'
+import { formatError, runSafe, runSafeAsync } from '../../stepFunctions/asl/utils/runner'
+
+namespace ResultLimitReachedNotification {
+    export const type: NotificationType<string, any> = new NotificationType('ssm/resultLimitReached')
+}
+
+namespace ForceValidateRequest {
+    export const type: RequestType<string, JsonLS.Diagnostic[], any, any> = new RequestType('ssm/validate')
+}
+
+// Create a connection for the server
+const connection: IConnection = createConnection()
+
+process.on('unhandledRejection', (e: any) => {
+    console.error(formatError('Unhandled exception', e))
+})
+process.on('uncaughtException', (e: any) => {
+    console.error(formatError('Unhandled exception', e))
+})
+
+console.log = connection.console.log.bind(connection.console)
+console.error = connection.console.error.bind(connection.console)
+
+const workspaceContext = {
+    resolveRelativePath: (relativePath: string, resource: string) => {
+        return URL.resolve(resource, relativePath)
+    },
+}
+
+// create the JSON language service
+let languageService = getLanguageService({
+    workspaceContext,
+    contributions: [],
+    clientCapabilities: JsonLS.ClientCapabilities.LATEST,
+})
+
+// Create a text document manager.
+const documents = new TextDocuments(JsonLS.TextDocument)
+
+// Make the text document manager listen on the connection
+// for open, change and close text document events
+documents.listen(connection)
+
+let clientSnippetSupport = false
+let dynamicFormatterRegistration = false
+let hierarchicalDocumentSymbolSupport = false
+
+let foldingRangeLimitDefault = Number.MAX_VALUE
+let foldingRangeLimit = Number.MAX_VALUE
+let resultLimit = Number.MAX_VALUE
+
+// After the server has started the client sends an initialize request. The server receives
+// in the passed params the rootPath of the workspace plus the client capabilities.
+connection.onInitialize(
+    (params: InitializeParams): InitializeResult => {
+        languageService = getLanguageService({
+            workspaceContext,
+            contributions: [],
+            clientCapabilities: params.capabilities,
+        })
+
+        // tslint:disable: no-unsafe-any
```

Comment:
What is this disabling?

Reply:
That is for the function `getClientCapability`